src/popup/components/Protocol/EnsureUserSavedMnemonic.tsx

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+  VerticalSequence,
+  Title,
+  Subtitle,
+  ClickableText,
+} from "@popup/components/Protocol/common";
+import React, { useEffect } from "react";
+import styled from "styled-components";
+import { getProtocolOptIn } from "@services/Factory";
+const ButtonContainer = styled.div`
+  width: 100%;
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  grid-column-gap: 20px;
+  grid-row-gap: 20px;
+  justify-items: stretch;
+  align-items: inital;
+`;
+
+const Button = styled.button`
+  cursor: pointer;
+  text-align: center;
+  color: var(--c-white);
+  background: var(--c-orange);
+  border-radius: var(--s-12);
+  padding: var(--s-10) var(--s-24);
+  font-weight: bold;
+  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out;
+  :disabled {
+    opacity: 0.4;
+    cursor: default;
+  }
+  :hover {
+    background: var(--c-dark-orange);
+  }
+`;
+
+interface CheckButton {
+  word: string;
+  isEnabled: boolean;
+}
+
+const WordButton = (props: {
+  word: string;
+  isEnabled: boolean;
+  onClick: () => void;
+}) => {
+  return (
+    <Button disabled={!props.isEnabled} onClick={props.onClick}>
+      {props.word}
+    </Button>
+  );
+};
+
+export function EnsureUserSavedMnemonic(props: { onComplete: () => void }) {
+  let [buttons, setButtons] = React.useState<CheckButton[]>([]);
+  const [mnemonicArr, setMnemonicArr] = React.useState<string[]>([]);
+
+  const [counter, setCounter] = React.useState(0);
+
+  useEffect(() => {
+    async function getMnemonic() {
+      const mnemonic = await getProtocolOptIn().getMnemonic();
+      setMnemonicArr((mnemonic as string).split(" "));
+      const sortedMnemonic = (mnemonic as string).split(" ").sort();
```

Comment:
This wouldn't need to be in a useEffect if the mnemonic was passed in. Also, if we change where the mnemonic comes from, this would have to change; while it wouldn't need to change if the mnemonic were just passed in.

Please pass in the mnemonic to verify as an argument to the component.

```tsx
function EnsureUserSavedMnemonic({mnemonic, onComplete}: {mnemonic: string, onComplete: () => void}) { ... }

<EnsureUserSavedMnemonic mnemonic={getProtocolOptIn().getMnemonic()} onComplete={...} />
```